.github/scripts/build-perf.js

```diff
@@ -0,0 +1,168 @@
+
+  page.on("requestfailed", (err) => {
+    console.log("\x1b[41m\x1b[30mERROR - requestfailed in puppeteer:", err);
+  });
+};
+
+/**
+ * Configure how the page is set up.
+ * The height and widgeth of the viewport can affect performance
+ * @private
+ * @param {Page} page An instance of puppeteer's Page
+ */
+const pageSetup = async (page) => {
+  // Setting a fixed viewport size lets us standardize all the perf tests
+  await page.setViewport({ width: 1280, height: 720 });
+
+  // Timestamp for tracking the internal performance of 'this' script
+  performanceMarkStart = performance.now();
+
+  // Track the http response object
+  page.on("response", addResponseSize);
+};
+
+/**
+ * Retrieve and calculate page metrics
+ * @private
+ * @param {Page} page An instance of puppeteer's page
+ * @param {string} sampleName
+ * @returns {object} Performance information
+ */
+const capturePageMetrics = async (page, sampleName) => {
+  const rawPerfEntries = await page.evaluate(() => {
+    return JSON.stringify(window.performance.getEntries());
+  });
+
+  const allPerformanceEntries = JSON.parse(rawPerfEntries);
+
+  // This number is derived from the timestamp of the last http request reported by puppeteer
+  // This is not reflective of when the app becomes usable. It's just the very last http request.
+  const elapsedRuntimeMS = Math.round(
+    allPerformanceEntries[allPerformanceEntries.length - 1].startTime +
+      allPerformanceEntries[allPerformanceEntries.length - 1].duration
+  );
+
+  // Finalize the self.performance numbers for this script
+  performanceMarkEnd = performance.now();
+
+  const totalScriptTimeMS = Math.round(performanceMarkEnd - performanceMarkStart);
+  const pageMetrics = await page.metrics();
+  const JSHeapUsedSizeBytes = pageMetrics.JSHeapUsedSize;
+
+  /**
+   * sampleName - name of the sample or bundle
+   * elapsedRuntimeMS - runtime in milliseconds derived from the applications last HTTP request
+   * pageTotalBytes - total number of bytes calculated using the http response object
+   * JSHeapUsedSizeBytes - JSHeapUsedSize as reported by puppeteer
+   * totalScriptTimeMS - approximate internal runtime of the library script in milliseconds.
+   * Useful for comparing against the `elapsedRuntimeMS`. Should not be used as an indicator of
+   * application performance, it's most useful for troubleshooting.
+   */
+  return {
+    sampleName,
+    elapsedRuntimeMS,
+    pageTotalBytes,
+    JSHeapUsedSizeBytes,
+    totalScriptTimeMS
+  };
+};
+
+/**
+ * Start the webserver
+ * @private
+ * @param {string} path
+ * @param {number} port
+ * @returns WebServer WebServer instance
+ */
+const startWebServer = (path, port) => {
+  webserver = new WebServer(path, port);
+  webserver.start();
+};
+
+/**
+ * Run browser performance tests
+ * @param {string} path
+ * @param {string} sampleInfo
+ * @returns {object} Object containing the performance metrics
+ */
+const browserPerformanceTest = async (path, sampleName = "") => {
```

Comment:
What's the reason for setting an empty string as the default for `sampleName`?

Reply:
To prevent a null check?

Reply:
Yep, prevent a null check